drud/ddev
# Fix apache-fpm behavior with php7.3, fixes #1346

## The Problem/Issue/Bug:

OP #1346: The ddev-webserver didn't come up at all when configured with apache-fpm and php7.3. This was due to missing apache config for php7.3

## How this PR Solves The Problem:

* Add in the missing files
* Add a full complement of nginx/apache/php tests into the container test.
* Bump the web container version number
* Remove symlinks that were checked in for the apache section of the web container. These didn't get brought into the docker build right (they came in a symlinks to paths with "\" in them). Previously we hadn't really been testing the Apache configurations, and this PR adds tests for every permutation of apache and php, so it uncovered the Windows build problem.   (This is what all the removals in containers/ddev-webserver/files/etc/apache2/conf-enabled are about, they're replaced with `a2enmod` command in start.sh.)
* Minor change to sanetestbot.sh to resolve a problem where it couldn't mount a directory which was a symlink.
* Added .gitattributes to explicitly control line endings in this repo; they never need to be anything but \n. The creeping-in of wrong linefeeds was caught in another PR. (I tried to solve this in a separate PR but ended up having to transplant it here to get a clean build)
* Remove "indexes" from apache config, since it should *not* be indexing directories with no index.php/index.html. This broke one of the tests.

## Manual Testing Instructions:

* Set your config.yaml for php7.3 and apache-fpm
* Configure your .ddev/config.yaml with `webimage: drud/ddev-webserver:20181221_php73_apache`
* `ddev start` should bring it up successfully.

## Automated Testing Overview:

This adds container-level tests for the full matrix of php vs webserver versions.

## Related Issue Link(s):

OP #1346 

## Release/Deployment notes:
